core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStoreMeta.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.page;
+
+import alluxio.DefaultStorageTierAssoc;
+import alluxio.StorageTierAssoc;
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageMetaStore;
+import alluxio.client.file.cache.store.PageStoreDir;
+import alluxio.collections.Pair;
+import alluxio.exception.PageNotFoundException;
+import alluxio.resource.LockResource;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.BlockStoreMeta;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimaps;
+import com.google.common.collect.Streams;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Metadata of the paged block store.
+ */
+public class PagedBlockStoreMeta implements BlockStoreMeta {
+  public static final String DEFAULT_TIER = "DefaultTier";
+  public static final String DEFAULT_DIR = "DefaultDir";
+  public static final String DEFAULT_MEDIUM = "SSD";
+  public static final DefaultStorageTierAssoc DEFAULT_STORAGE_TIER_ASSOC =
+      new DefaultStorageTierAssoc(ImmutableList.of(DEFAULT_MEDIUM));
+
+  private final long mCapacity;
+  private final long mUsed;
+  private final int mNumBlocks;
+  private final List<String> mDirPaths;
+  private final List<Long> mCapacityBytesOnDirs;
+  private final List<Long> mUsedBytesOnDirs;
+  @Nullable
+  private final Map<BlockStoreLocation, List<Long>> mBlockLocations;
+  @Nullable
+  private final List<Long> mBlocks;
```

Comment:
I might prefer to use Optional instead of Nullable

Reply:
Fixed in [79a8669](https://github.com/Alluxio/alluxio/pull/15903/commits/79a86697f04012a20853b5baadd07bff4e5f8316)